runtime/pkg/rduckdb/db.go

```diff
@@ -111,87 +111,49 @@ func (d *DBOptions) ValidateSettings() error {
 			return fmt.Errorf("unable to parse max_memory: %w", err)
 		}
 
-		read.MaxMemory = fmt.Sprintf("%d bytes", int64(bytes)/2)
-		write.MaxMemory = fmt.Sprintf("%d bytes", int64(bytes)/2)
-	}
-
-	if read.MaxMemory == "" != (write.MaxMemory == "") {
-		// only one is defined
-		var mem string
-		if read.MaxMemory != "" {
-			mem = read.MaxMemory
-		} else {
-			mem = write.MaxMemory
-		}
-
-		bytes, err := humanReadableSizeToBytes(mem)
-		if err != nil {
-			return fmt.Errorf("unable to parse max_memory: %w", err)
-		}
-
-		read.MaxMemory = fmt.Sprintf("%d bytes", int64(bytes)/2)
-		write.MaxMemory = fmt.Sprintf("%d bytes", int64(bytes)/2)
+		memoryLimitBytes = int64(bytes)
 	}
 
-	var readThread, writeThread int
-	if read.Threads != "" {
-		readThread, err = strconv.Atoi(read.Threads)
-		if err != nil {
-			return fmt.Errorf("unable to parse read threads: %w", err)
-		}
-	}
-	if write.Threads != "" {
-		writeThread, err = strconv.Atoi(write.Threads)
+	threads := d.CPU
+	if threads == 0 {
+		db, err := sql.Open("duckdb", "")
 		if err != nil {
-			return fmt.Errorf("unable to parse write threads: %w", err)
-		}
-	}
-
-	if readThread == 0 && writeThread == 0 {
-		connector, err := duckdb.NewConnector("", nil)
-		if err != nil {
-			return fmt.Errorf("unable to create duckdb connector: %w", err)
+			return err
 		}
-		defer connector.Close()
-		db := sql.OpenDB(connector)
 		defer db.Close()
 
 		row := db.QueryRow("SELECT value FROM duckdb_settings() WHERE name = 'threads'")
-		var threads int
 		err = row.Scan(&threads)
 		if err != nil {
 			return fmt.Errorf("unable to get threads: %w", err)
 		}
-
-		read.Threads = strconv.Itoa((threads + 1) / 2)
-		write.Threads = strconv.Itoa(threads / 2)
 	}
 
-	if readThread == 0 != (writeThread == 0) {
-		// only one is defined
-		var threads int
-		if readThread != 0 {
-			threads = readThread
+	if d.ReadWriteResourceSplitRatio == 0 {
+		d.ReadSettings["memory_limit"] = fmt.Sprintf("%d bytes", memoryLimitBytes/2)
+		d.WriteSettings["memory_limit"] = fmt.Sprintf("%d bytes", memoryLimitBytes/2)
+		if threads < 2 {
+			d.ReadSettings["threads"] = "1"
 		} else {
-			threads = writeThread
+			d.ReadSettings["threads"] = strconv.Itoa(threads / 2)
 		}
+		d.WriteSettings["threads"] = strconv.Itoa((threads + 1) / 2)
+	} else {
```

Comment:
Isn't this the same as having the ration be `0.5` by default? Given the comment above, perhaps it's easiest to just have the `drivers/duckdb` package set the ratio to 0.5 in `DBOptions` by default.